### ISSUE TITLE
New rule proposal: no-nonoctal-decimal-escape

### ISSUE THREAD

**Issue by mdjermanovic** [eslint/eslint#13765]
**Please describe what the rule should do:**

Disallow `\8` and `\9` escape sequences in string literals.

**What new ECMAScript feature does this rule relate to?**

[NonOctalDecimalEscapeSequence](https://tc39.es/ecma262/#prod-annexB-NonOctalDecimalEscapeSequence) has been recently added to the specification:

https://github.com/tc39/ecma262/pull/2054

**What category of rule is this? (place an "X" next to just one item)**

[ ] Warns about a potential error (problem)
[X] Suggests an alternate way of doing something (suggestion)
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule will warn about:**

```js
"\8"

"\9"
```

**Why should this rule be included in ESLint (instead of a plugin)?**

It disallows the use of an Annex B feature:

> These features are not considered part of the core ECMAScript language. Programmers should not use or assume the existence of these features and behaviours when writing new ECMAScript code.

**Are you willing to submit a pull request to implement this rule?**

Yes.

Previous discussion about including this in some of the existing rules: https://github.com/eslint/eslint/issues/13740

**Comment by nzakas** [comment]
Marking as accepted as there is a champion and three :+1:s

**Comment by mdjermanovic** [comment]
I'll work on this.
